### DISCUSSION TITLE
Using persistent layouts and data from an endpoint

### DISCUSSION THREAD

**Discussion by samynaj** [vercel/next.js#18231] [Help]
Hi, so I have a project am working on and I have different layouts for different groups of pages. These layouts have need for user data from a backend. Problem is I am currently fetching user data on every page and passing it down as props to the layouts which I feel is a bad practice even though it works. I need a way to fetch user data just once each for the different layouts. I just read about persistent layout but I can't seem to put the pieces together to achieve my goal.

**Comment by abhinavsingi** [comment]
Would this help?
https://adamwathan.me/2019/10/17/persistent-layout-patterns-in-nextjs/
